rhoshift/utils/utils.py

```
@@ -0,0 +1,360 @@
+
+            if output_mode == OutputMode.STREAM:
+                return await _run_command_streaming(cmd, timeout, cwd, env, shell, log_output, start_time)
+            else:
+                return await _run_command_buffered(cmd, timeout, cwd, env, shell, log_output, output_mode, start_time)
+
+        except asyncio.TimeoutError:
+            if attempt <= max_retries:
+                logging.warning(f"Command timed out, retrying in {retry_delay} seconds...")
+                await asyncio.sleep(retry_delay)
+            else:
+                raise CommandTimeoutError(f"Command timed out after {timeout} seconds: {cmd}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)